swarm/animator.py

```diff
@@ -0,0 +1,42 @@
+#  -*- coding: utf-8 -*-
+__author__ = "Varun Nayyar <nayyarv@gmail.com>"
+
+from matplotlib import pyplot as plt, animation
+import seaborn as sns
+
+plt.rcParams["figure.figsize"] = (14.0, 7.0)
+sns.set()
+
+
+def make_animation(xd, yd, data, title, destfile):
+    # Determine a proper format
+    nepoch = data[0].shape[0]
+    fig = plt.figure()
+    ax = plt.axes()
+    plt.title(title)
+    ax.plot(xd, yd, ".")
+
+    line_ref = []
+    for i in range(len(data)):
+        (liner,) = ax.plot([], [], lw=2)
+        line_ref.append(liner)
+
+    def init():
+        for line in line_ref:
+            line.set_data([], [])
+        return line_ref
+
+    def animate(i):
+        for dnum, line in enumerate(line_ref):
+            line.set_data(xd, data[dnum][i])
+        return line_ref
+
```

Comment:
part of me wonders whether the steps up to this point to build the plot couldn't be wrapped up into a separate function?